### PR TITLE
Add role_title and addedby_username in project memberships and usergroup api

### PR DESCRIPTION
**Pull request by k9845**
Changed API:
- for project-memberships from `/api/v1/project-memberships` to `/api/v1/projects/{project.id}/project-memberships/`
- for project-usergroups from `/api/v1/project-usergroups` to `/api/v1/projects/{project.id}/project-usergroups/`

CC: @samshara  @subinasr

### CODE REVIEW

**Review by annshress — APPROVED** [review]

File: apps/project/serializers.py [hunk 1]
```diff
@@ -147,18 +149,24 @@ def validate_project(self, project):
...
+        data['project_id'] = int(self.context['view'].kwargs['project_id'])
+        # get the members of the project
+        members = ProjectMembership.objects.filter(
+            project=data['project_id']
+        ).values('member')
+        member = data.get('member')
+        if member:
+            member = member.id
+        if member in [mem['member'] for mem in members]:
+            raise serializers.ValidationError({'member': 'Member already exist'})
```
Comment (annshress) [inline apps/project/serializers.py]: Separate this validation to a different function.

Further, use something like below,

```
if ProjectMembership.objects.filter(project=data['project_id']).filter(member=data.get('member')).exists():
    raise ...
```

File: apps/project/serializers.py [hunk 2]
```diff
@@ -128,10 +128,12 @@ class ProjectMembershipSerializer(RemoveNullFieldsMixin,
         read_only=True,
         many=True,
     )
+    role_title = serializers.CharField(source='role.title', read_only=True)
```
Comment (thenav56) [inline apps/project/serializers.py]: Let's define SimpleRoleSerializer and use that instead.

role_details = SimpleRoleSerializer(source='role', read_only=True)

Make sure to also use select related. So role_id and role_title should be enough.

File: apps/project/serializers.py [hunk 3]
```diff
@@ -128,10 +128,12 @@ class ProjectMembershipSerializer(RemoveNullFieldsMixin,
...
 
     class Meta:
         model = ProjectMembership
         fields = '__all__'
+        read_only_fields = ('project', )
```
Comment (thenav56) [inline apps/project/serializers.py]:
```suggestion
        read_only_fields = ('project',)
```

File: apps/project/serializers.py [hunk 4]
```diff
@@ -446,10 +451,17 @@ def create(self, validated_data):
 
 class ProjectUserGroupSerializer(serializers.ModelSerializer):
     title = serializers.CharField(source='usergroup.title', read_only=True)
+    role_title = serializers.CharField(source='role.title', read_only=True)
```
Comment (thenav56) [inline apps/project/serializers.py]: Let's define SimpleRoleSerializer and use that instead.

role_details = SimpleRoleSerializer(source='role', read_only=True)

Make sure to also use select related. So role_id and role_title should be enough.

**Review by thenav56 — APPROVED** [review]

File: apps/project/serializers.py (hunk 1 above)
Comment (k9845) [reply apps/project/serializers.py]: Done!

**Review by annshress — APPROVED** [review]

**Pull request merged**